controllers/protectedvolumereplicationgrouplist_controller.go

```diff
@@ -50,31 +50,33 @@ type ProtectedVolumeReplicationGroupListInstance struct {
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.9.2/pkg/reconcile
 func (r *ProtectedVolumeReplicationGroupListReconciler) Reconcile(ctx context.Context, req ctrl.Request,
 ) (ctrl.Result, error) {
-	log := ctrl.Log.WithName("controllers").WithName("protectedvolumereplicationgrouplist").
-		WithValues("name", req.NamespacedName.Name)
-
-	log.Info("protectedvolumereplicationgrouplist reconcile start")
-
 	// save all the commonly used parameters in a struct
 	s := ProtectedVolumeReplicationGroupListInstance{
 		reconciler: r,
 		ctx:        ctx,
-		log:        log,
+		log:        ctrl.Log.WithName("pvrgl").WithValues("name", req.NamespacedName.Name),
 		instance:   &ramendrv1alpha1.ProtectedVolumeReplicationGroupList{},
 	}
 
+	s.log.Info("reconcile start")
+	defer s.log.Info("reconcile end")
+
 	// get ProtectedVolumeReplicationGroupListInstance and save to s.instance
 	if err := r.Client.Get(ctx, req.NamespacedName, s.instance); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(fmt.Errorf("get: %w", err))
 	}
 
+	// add the instance's UID and the resource version to the log
+	s.log = s.log.WithValues("UID", s.instance.ObjectMeta.UID, "version", s.instance.ObjectMeta.ResourceVersion)
+	s.ctx = ctrl.LoggerInto(ctx, s.log)
+
 	if s.instance.Status != nil {
 		return ctrl.Result{}, nil
 	}
 
 	// get target profile from spec
 	s3profileName := s.instance.Spec.S3ProfileName
-	log.Info(fmt.Sprintf("targetProfileName=%s", s3profileName))
+	s.log.Info(fmt.Sprintf("targetProfileName=%s", s3profileName))
 
```

Comment:
Not related to your change but this log is not very useful. The s3profile name is in the resource and repeating it in the log is not helpful in normal conditions. I can be useful when we report errors.

Reply:
Ack, will do another PR to address the unwanted logs.